controllers/toolchaincluster_resources/toolchaincluster_resources_controller.go

```diff
@@ -0,0 +1,68 @@
+package toolchaincluster_resources
+
+import (
+	"context"
+	"embed"
+	"fmt"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	applycl "github.com/codeready-toolchain/toolchain-common/pkg/client"
+	"github.com/codeready-toolchain/toolchain-common/pkg/template"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&toolchainv1alpha1.ToolchainCluster{}).
+		Complete(r)
+}
+
+// Reconciler reconciles a ToolchainCluster object
+type Reconciler struct {
+	client    runtimeclient.Client
+	scheme    *runtime.Scheme
+	templates *embed.FS
+}
+
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io;authorization.openshift.io,resources=rolebindings;roles;clusterroles;clusterrolebindings,verbs=*
+//+kubebuilder:rbac:groups="",resources=serviceaccounts,verbs=get;list;watch;create;update;patch;delete
```

Comment:
same here concern here about the permissions needed to apply the objects

Reply:
TBH, I'm not sure if the kubebuilder reads the content from the dependencies. In other words, I don't think that this will have any result on the manifest generated by the kubebuilder. 
This should be rather placed in the operator code - we do that also with the ToolchainCluster controller (and other stuff), the marker is added to the main.go:  https://github.com/codeready-toolchain/host-operator/blob/f27b2b90c5132b3346a8eb13254307fb209c46d4/main.go#L92-L94